CommunityToolkit.Uwp.Authentication/WindowsProvider.cs

```diff
@@ -0,0 +1,453 @@
+        private const string GraphResourcePropertyKey = "resource";
+        private const string GraphResourcePropertyValue = "https://graph.microsoft.com";
+        private const string MicrosoftAccountAuthority = "consumers";
+        private const string MicrosoftProviderId = "https://login.microsoft.com";
+        private const string SettingsKeyAccountId = "WindowsProvider_AccountId";
+        private const string SettingsKeyProviderId = "WindowsProvider_ProviderId";
+
+        // Default/minimal scopes for authentication, if none are provided.
+        private static readonly string[] DefaultScopes = { "User.Read" };
+
+        /// <summary>
+        /// Gets the client id obtained from Azure registration.
+        /// </summary>
+        public string ClientId => _clientId;
+
+        /// <summary>
+        /// Gets the list of scopes to pre-authorize during authentication.
+        /// </summary>
+        public string[] Scopes => _scopes;
+
+        /// <summary>
+        /// Gets configuration values for the AccountsSettingsPane.
+        /// </summary>
+        public AccountsSettingsPaneConfig? AccountsSettingsPaneConfig => _accountsSettingsPaneConfig;
+
+        /// <summary>
+        /// Gets a cache of important values for the signed in user.
+        /// </summary>
+        protected IDictionary<string, object> Settings => ApplicationData.Current.LocalSettings.Values;
+
+        private string _clientId;
+        private string[] _scopes;
+        private WebAccount _webAccount;
+        private AccountsSettingsPaneConfig? _accountsSettingsPaneConfig;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WindowsProvider"/> class.
+        /// </summary>
+        /// <param name="clientId">Registered ClientId. You can access an acount without one, but any Graph requests will fail.</param>
+        /// <param name="scopes">List of Scopes to initially request.</param>
+        /// <param name="accountsSettingsPaneConfig">Configuration values for the AccountsSettingsPane.</param>
+        public WindowsProvider(string clientId = "", string[] scopes = null, AccountsSettingsPaneConfig? accountsSettingsPaneConfig = null)
+        {
+            _clientId = clientId;
+            _scopes = scopes ?? DefaultScopes;
+            _accountsSettingsPaneConfig = accountsSettingsPaneConfig;
+
+            _webAccount = null;
+
+            State = ProviderState.SignedOut;
+        }
+
+        /// <inheritdoc />
+        public override async Task AuthenticateRequestAsync(HttpRequestMessage request)
+        {
+            string token = await GetTokenAsync();
+            request.Headers.Authorization = new AuthenticationHeaderValue(AuthenticationHeaderScheme, token);
+        }
+
+        /// <inheritdoc />
+        public override async Task LoginAsync()
+        {
+            if (_webAccount != null || State != ProviderState.SignedOut)
+            {
+                await LogoutAsync();
```

Comment:
Alternatively, could we NOP if the user is signed in already? If a developer calls LoginAsync while a user is signed in, I'd expect the method to do nothing.

Reply:
Good point! I'll fix this